assets/javascripts/discourse/lib/block-kit.js

```diff
@@ -0,0 +1,70 @@
+function stringToDOMNode(string) {
+  let template = document.createElement("template");
+  string = string.trim();
+  template.innerHTML = string;
+  return template.content.firstChild;
+}
+
+export default class BlockKit {
+  static decorate(element) {
+    return new this(element).init();
+  }
+
+  constructor(element) {
+    this.element = element;
+  }
+
+  init() {
+    this.element
+      .querySelectorAll(".d-wrap[data-wrap=blocks]")
+      .forEach((blocks) => {
+        blocks.querySelectorAll(".d-wrap[data-wrap=block]").forEach((block) => {
+          const { type } = block.dataset;
+
+          if (type === "context") {
```

Comment:
Do we have all these as separate conditions even though they call the same function for future flexibility? Just curious

Reply:
Yes for this demo I just didn't  want to think about this.